packages/ui-components/src/components/vdom.ts

```diff
@@ -72,22 +78,33 @@ export abstract class ReactWidget extends Widget {
   private renderDOM(): Promise<void> {
     return new Promise<void>(resolve => {
       const vnode = this.render();
+      if (this._rootDOM === null) {
+        this._rootDOM = createRoot(this.node);
+      }
       // Split up the array/element cases so type inference chooses the right
       // signature.
       if (Array.isArray(vnode)) {
-        ReactDOM.render(vnode, this.node, resolve);
+        this._rootDOM.render(vnode);
+        // Resolves after the widget has been rendered.
+        // https://github.com/reactwg/react-18/discussions/5#discussioncomment-798304
+        setTimeout(resolve, 0);
```

Comment:
> We may need to look for other ways to simulate event in the tests. I cannot figure out why this is happening with React 18 as I don't see mention about changes in the event handling.

About the JS tests (if this is the ones you're talking about), the error is that `ReactDOM.render` has been replaced in *react-dom 18*, and does not allow callback function anymore.

This [discussion](https://github.com/reactwg/react-18/discussions/5) explains it, but I didn't find a way to implement it for the `ReactWidget`. I tried the [workarounds](https://github.com/reactwg/react-18/discussions/5#discussioncomment-798304) without success.

It results in trying to click on the widget before it has been fully rendered. It can probably be fixed in tests by waiting for the button, but I don't know if there can be consequences elsewhere, as there will be no more promise to wait for.

Reply:
Thanks for the pointer. The best path is through the `ref` callback. But this is not acceptable for us (we cannot expect developer to set that for all `ReactWidget`). Could we try something more crude like waiting for `this.node` to have children?

Reply:
> The best path is through the `ref` callback

This is what is done in the `vdom` package, but it 's not working for `ReactWidget`.

 > Could we try something more crude like waiting for `this.node` to have children?

I'll try this